packages/bootleg-tokens/contracts/AbstractSharedRoyaltyToken.sol

```diff
@@ -0,0 +1,205 @@
+
+  constructor () public {
+    // register the supported interfaces to conform to ERC721 via ERC165
+    _registerInterface(_INTERFACE_ID_SRT);
+  }
+
+  /**
+   * @notice Transfers the ownership of a given token ID to another address
+   *   Usage of this method is discouraged, use `safeTransferFrom` whenever possible
+   *   Requires the msg sender to be the owner, approved, or operator
+   *
+   * @dev Only overriding this function as both safeTransferFrom functions
+   *   call this one
+   *
+   * @param from current owner of the token
+   * @param to address to receive the ownership of the given token ID
+   * @param tokenId uint256 ID of the token to be transferred
+   */
+  function transferFrom(address from, address to, uint256 tokenId) public payable {
+    super.transferFrom(from, to, tokenId);
+
+    _addFranchisor(to, msg.value, tokenId);
+  }
+
+  /**
+   * @notice Gets the franchisor balance of the specified address
+   *
+   * @param franchisor address to query the balance of
+   * @return uint256 representing the amount franchised by the passed address
+   */
+  function franchisorBalanceOf(address franchisor) public view notZeroAddr(franchisor) returns (uint256) {
+    return _franchisorTokensCount[franchisor];
+  }
+
+  /**
+   * @notice Withdraws payment accumulated from transfer of a given token from
+   *  the last withdrawn payment up to a count
+   *
+   * @dev if count is greater than the total number of payments left to withdraw,
+   *  the function will only withdraw up to the payments left
+   *
+   * @param tokenId The identifier for an NFT
+   * @param franchisor address to withdraw payment for
+   * @param count The number of payments to traverse
+   */
+  function withdrawPayment(address payable franchisor, uint256 count, uint256 tokenId) public notZeroAddr(franchisor) {
+    Token storage token = _tokens[tokenId];
+
+    uint256 withdrawPaymentsLeft = franchisorWithdrawPaymentsLeft(franchisor, tokenId);
+    uint256 lastWithdrawnIndex = token.franchisorNextWithdrawIndex[franchisor];
+    uint256 withdrawCount = lastWithdrawnIndex.add(count) > token.payments.length
+      ? withdrawPaymentsLeft
+      : count;
+
+    uint256 balance = paymentBalanceOf(
+      franchisor,
+      token.franchisorNextWithdrawIndex[franchisor],
+      withdrawCount,
+      tokenId
+    );
+
+    token.franchisorNextWithdrawIndex[franchisor] = lastWithdrawnIndex.add(withdrawCount);
+
+    franchisor.transfer(balance);
+
+    emit WithdrawnPayment(franchisor, balance);
+  }
+
+  /**
+   * @notice Withdraws remaining payment balance accumulated from transfer of a
+   *   given token
+   * @dev Escrow should keep track of the payments that have been processed to
+   *   avoid double withdrawals
+   *
+   * @param tokenId The identifier for an NFT
+   * @param franchisor address to withdraw payment for
+   */
+  function withdrawPayment(address payable franchisor, uint256 tokenId) public notZeroAddr(franchisor) {
```

Comment:
If this is available, why wouldn't someone want to call this version all the time? When would the `count` based one be used?... besides here.

Reply:
The only time is if this one fails b/c it exceeds the gas limit.

This will _likely_ happen if `count` is greater than 1000